### PR TITLE
docs: add glossary

### PR DESCRIPTION
**Pull request by deepeshaburse**
This PR adds a `Glossary` section in the documentation.

### CODE REVIEW

**Comment by deepeshaburse** [comment]
cc: @TerryHowe

File: docs/glossary.mdx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+The [OCI Distribution Spec](https://github.com/opencontainers/distribution-spec/blob/main/spec.md) defines an API protocol to facilitate and standardize the distribution of content.
+
+### Extensions
+
+The Distribution Specification Project includes a process and API for prototyping and testing [extensions](https://github.com/opencontainers/distribution-spec/blob/main/extensions/README.md) to the Distribution API.
+
+### Image Layout
+
+The [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) is the directory structure for OCI content-addressable blobs and location-addressable references (refs). It must contain a `blobs` directory, `oci-layout` file and an `index.json` file. Our how-to guide on OCI Layouts can help understand the concepts and usage better.
```
Comment (TerryHowe) [inline docs/glossary.mdx:46]:
```suggestion
The [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) is the directory structure for OCI content-addressable blobs and location-addressable references (refs). It must contain a `blobs` directory, `oci-layout` file and an `index.json` file. To learn more, check out the [how-to guide on OCI Layouts](link).
```
Not sure what the link is

**Review by TerryHowe — COMMENTED** [review]
I'll look at the rest later.

File: docs/glossary.mdx (hunk 1 above)
Comment (deepeshaburse) [reply docs/glossary.mdx:46]: The link isn't available yet. It's the guide in PR #227. I'll make the other change though.

File: docs/glossary.mdx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+The [OCI Image Spec](https://github.com/opencontainers/image-spec/blob/main/spec.md) defines an OCI Image, consisting of an image manifest, an image index (optional), a set of filesystem layers, and a configuration.
+
+### Image Index Specification
+
+The [image index](https://github.com/opencontainers/image-spec/blob/main/image-index.md) is a higher-level manifest which points to specific image manifests, ideal for one or more platforms. It is a multi-descriptor entry point.
+
+### Local Registry
+
+A registry is a place where container images and artifacts can easily be stored and accessed. Whereas, a local registry (like [zot](https://zotregistry.io/)) is a registry that is present on our local machine. You can follow our [quick start guide](./quickstart.mdx) if you would like to try using zot registry.
```
Comment (TerryHowe) [inline docs/glossary.mdx:62]:
```suggestion
A registry is a place where container images and artifacts can easily be stored and accessed. Whereas, a local registry (like [zot](https://zotregistry.io/)) is a registry that is running on our local machine. You can follow our [quick start guide](./quickstart.mdx) if you would like to try using a zot registry.
```

File: docs/glossary.mdx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+### Manifest Referrers API
+
+Artifact-manifest provides the ability to reference artifacts to existing artifacts. Reference artifacts include signatures, SBoMs and many other types. The [manifest `referrers` API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) returns all artifacts that have a `subject` of the given manifest digest.
```
Comment (TerryHowe) [inline docs/glossary.mdx:66]:
```suggestion
An artifact-manifest may reference artifacts including signatures, SBoMs and many other types. The [manifest `referrers` API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) returns all artifacts that have a `subject` of the given manifest digest.
```

File: docs/glossary.mdx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+### Referrer API
+
+The Referrers API returns a list of manifests that reference a blob. You can understand how to use this API by referring to the [details](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#listing-referrers) given in the distribution spec. 
+
+### Referrer Tag Schema
+
+The Referrer Tag Schema helps to pull the current list of referrers based on different digest formats or tags.
```
Comment (TerryHowe) [inline docs/glossary.mdx:74]: I'm not entirely sure here, but doesn't the schema define the supported format of digests or tags?

**Review by TerryHowe — DISMISSED** [review]
/lgtm

**Review by sajayantony — DISMISSED** [review]

File: docs/glossary.mdx (hunk 4 above)
Comment (deepeshaburse) [reply docs/glossary.mdx:103]: @TerryHowe accidentally missed this comment. That is what I understood after reading the content of opencontainers, however, I may have misinterpreted it.

**Comment by deepeshaburse** [comment]
@TerryHowe and @sajayantony I added semantic line breaks for increasing readability.

File: docs/glossary.mdx (hunk 4 above)
Comment (sajayantony) [reply docs/glossary.mdx:103]: Hope this might help  - https://github.com/oras-project/oras-www/pull/240

**Review by sajayantony — DISMISSED** [review]

File: docs/glossary.mdx [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+### Extensions
+
+The Distribution Specification Project includes a process and API for prototyping and testing [extensions](https://github.com/opencontainers/distribution-spec/blob/main/extensions/README.md) to the Distribution API.
+
+### Image Layout
+
+The [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) is the directory structure for OCI content-addressable blobs and location-addressable references (refs). 
+It must contain a `blobs` directory, `oci-layout` file and an `index.json` file. 
+To learn more, check out the how-to guide on OCI Layouts.
```
Comment (sajayantony) [inline docs/glossary.mdx:65]: Should add this once the other PR is complete?

Comment (deepeshaburse) [reply docs/glossary.mdx:65]: @sajayantony yes, we should wait for that PR to merge.

Comment (TerryHowe) [reply docs/glossary.mdx:65]: Which PR are we waiting for here?

Comment (deepeshaburse) [reply docs/glossary.mdx:65]: @TerryHowe didn't notice the PR got merged, I'll update the link in the glossary and it should be ready to merge.

**Comment by deepeshaburse** [comment]
@TerryHowe @sajayantony the PR is ready to be merged if the content looks fine.

**Review by sajayantony — APPROVED** [review]

**Pull request merged**